### ISSUE TITLE
Popup does not always reposition on scroll

### ISSUE THREAD

**Issue by Xizario** [telerik/kendo-react#93]
Affects DropDown or DatePicker inside the filter row of scrollable Grid.
https://stackblitz.com/edit/react-ughtbz-zt6zer

**Comment by negreamihneavlad** [comment]
Can this be fixed using multiple solutions?
1. If popup opened, disable scrolling until is closed.
2. If popup opened & table is scrolled, then automatically close the popup.
3. Move popup along with table if scrolled.

**Comment by Xizario** [comment]
Scrolling is not something that can be disabled, since it is browser feature and is async in most of the cases.

Closing the popup is valid solution, but will cause problems when there are multiple popups, and it could cause closing popup that should remain opened. Nested popups for example happen with Menu (it uses the popup) that have template with Dropdowns inside it.

We are going with the third option, recalculate the position. It already works for most of the cases, the issue with the header of the grid since the header is not directly scrolled, the grid's body is actually scrolled and the header is synced to the body.

**Comment by negreamihneavlad** [comment]
For the first option, blocking the overflow property on table is not an option? This is the one who makes it scrollable.
For the second one, regarding the filters from the table, I see that you can only have one opened at a time. If I have a datepicker open and then I try to open a select/dropdown, the datepicker will close.

I would love to have one of the first two options in place, because my only concern for the third one is that it will feel a bit laggy(recalculating the popup position at each scroll event).

**Comment by Xizario** [comment]
I wrote the above by assuming real-life application would have scrollable containers that are user defined or part or other libraries, and we don't have control over them, nor we should modify them. Changing the overflow will most likely break the layout, cause repaint and in the end being significantly slower than reposition of the popup.
Components like the Dialog or Window which could be modal are having such functionality: preventing you from doing anything else when they are opened.

The DatePicker closes when it is blurred by default, but can be [forced](https://www.telerik.com/kendo-react-ui/components/dateinputs/datepicker/controlled-state/#toc-controlling-the-popup-state) to stay opened. The Popup is stand alone component, even if we use it internally, it can be used in various of ways, so we can't relay that there will be single popup in someones application, and they can also nest them if needed.

I see your concern there, we will do our best to keep the performance at top level.

**Comment by nstoychev** [comment]
We have published a fix in our dev channel: https://www.npmjs.com/package/@progress/kendo-react-grid/v/2.2.0-dev.201810091021. Please refer to the [this article](https://www.telerik.com/kendo-react-ui/components/installation/development-builds/) on how to obtain the development builds.

**Comment by negreamihneavlad** [comment]
I just installed it. It changes its position now, but what I was afraid of, it happens. The table gets very heavy on scrolling and lagging pretty hard. And if the dropdown input goes out of screen (the table container has fixed width), the popup keeps moving until it hits the sides of the view.

Again, it would be really nice to close the popup on scrolling. To not interfere with others components, adding a prop boolean on component would be an option (e.g. closeOnScroll, or on <Column closeFilterPopupOnScroll />)

Thanks for your effort.

**Comment by Xizario** [comment]
After further discussions and tests we reverted the old behavior: the popup does not reposition. Main reasons:
When device have small display, like an mobile device, the user is constantly scrolling and zooming and the auto-reposition is causing troubles.
When the anchor is not visible, the popup is positioned outside of the are where it should be placed.

For closing the popups on scroll we have exposed onScroll event of the grid that can be used for this particular case
